source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,101 @@ Find a Document
+first document. Another example of an option is a **projection**, which
+specifies or restricts the fields in your returned document. You can specify
+options by passing the ``FindOne()`` method a pointer to a ``FindOneOptions`` struct.  
+You can configure a ``FindOneOptions`` struct by using setter methods such as the
+``FindOneOptions.SetProjection()`` method. For more information on the sort option,
+:doc:`see our guide on sorting </fundamentals/crud/read-operations/sort/>`.
+For more information on the projection option, 
+:doc:`see our guide on projections </fundamentals/crud/read-operations/project/>`.  
+
+..
+   return
+
+If the query filter you specified in your ``FindOne()`` method call matches one
+or more documents in your collection, it returns the first matched result. The
+return type ``SingleResult`` represents a single document returned from a query
+to a MongoDB instance. You can call the ``SingleResult.Decode()`` method
+to unmarshal a ``SingleResult`` struct type into a variable. Pass the ``Decode()`` 
+method a pointer to the variable that you would like to contain your returned
+document. 
+
+..
+   error 
+
+When you call the ``Decode()`` method to unmarshal your result, the method
+returns an ``error`` interface type which can contain one of the following
+values: 
+
+- ``nil`` if a document matched your query, and there were no errors retrieving
+  the document. 
+- ``ErrNoDocuments`` if no documents matched your query.
+- If the driver retrieved your document but could not unmarshal your result, the
+  ``Decode()`` method returns the unmarshaling error.
+- If there was an error retrieving your document during execution of the
+  ``FindOne()`` method, the error propagates to the ``Decode()`` method and
+  the ``Decode()`` method returns the error. 
+
+Example
+-------
+
+The following code example retrieves a single document from the ``movies``
+collection. It uses the following arguments for the ``FindOne()`` method:
+
+- An empty **Context** instance returned from the ``context.TODO()`` function.
+- A **filter document** that only matches movies with the title exactly matching
+  the text ``"The Room"``.
+- **Options** that specify a **sort** of matched documents in
+  descending order by rating, such that the first document has the highest
+  rating. The options also specify a **projection** for the results so
+  that the MongoDB instance only sends the ``title`` and ``imdb`` fields of
+  the retrieved document.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: findOne.go
+   :language: go
+
+If you run the example above, you should see a result that looks like this:
+
+.. literalinclude:: findOne-output.txt
+   :language: none
+
+.. include:: /includes/sample-data-note.rst
+
+For more information on the methods, interfaces, structs, and functions
+mentioned on this page, see the following API documentation:
+
+- :go-api:`*Collection.FindOne() <mongo#Collection.FindOne>`
```

Comment:
Is the prefix asterisk a standard way of referring to Go methods with a pointer receiver? It does look a bit odd to me, but I may not have full context.

As evidence to the contrary, I see a few [examples within the Go driver docs](https://github.com/mongodb/mongo-go-driver/blob/b47cf2ee6403a34cec04aebe5bdf82dfab2bba74/mongo/index_view.go#L40:L40) without the asterisk when referring to a type's method.

cc @iwysiu in case you know off-hand.

Reply:
Thanks for the note. We were trying to come up with a convention as a team but we were not sure what was best. In the [Donavon and Kernighan Go Book](https://www.gopl.io/) they use `(*<reciever name>).method()` name on page 158, 164, and 167. We had a discussion about this and felt it was somewhat hard to read, and we had trouble finding this notation used elsewhere, so we decided to try the `*<method name>.method()` as it is somewhat similar to the way it is discussed [in a tour of go](https://tour.golang.org/methods/4) and seemed easier to read.

I think recently we have been leaning more towards dropping the `*` altogether, as it seems like in Go most people do not distinguish between pointer receiver methods and value receiver methods in prose.

We are not at all attached to this notation and are totally open to any suggestions. If you Kevin or Isabella have a preference either way we would appreciate hearing your thoughts.

Reply:
Thanks for sharing your thoughts and for hunting down such a relevant example. After having a brief conversation around this thread in the DevEd DBX Slack Channel, we have decided to remove the `*` in links to pointer receiver methods. 

Fix: https://github.com/mongodb/docs-golang/pull/2/commits/2891c941765fccfbf603259edf55d0ef38d26e64

Going forward, we plan to not have a `*` when linking to the API docs for pointer receiver methods. @iwysiu, please feel free to let us at the DevEd DBX team know if you disagree with this stance and would like us to structure these links differently.